src/Symfony/Component/Messenger/Tests/DependencyInjection/MessengerPassTest.php

```diff
@@ -262,14 +263,42 @@ public function testRegistersMiddlewareFromServices()
         $container = $this->getContainerBuilder();
         $container->register($fooBusId = 'messenger.bus.foo', MessageBusInterface::class)->setArgument(0, array())->addTag('messenger.bus');
         $container->register('messenger.middleware.allow_no_handler', AllowNoHandlerMiddleware::class)->setAbstract(true);
+        $container->register('middleware_with_factory', UselessMiddleware::class)->addArgument('some_default')->setAbstract(true);
+        $container->register('middleware_with_factory_using_default', UselessMiddleware::class)->addArgument('some_default')->setAbstract(true);
         $container->register(UselessMiddleware::class, UselessMiddleware::class);
 
-        $container->setParameter($middlewareParameter = $fooBusId.'.middleware', array(UselessMiddleware::class, 'allow_no_handler'));
+        $container->setParameter($middlewareParameter = $fooBusId.'.middleware', array(
+            array('id' => UselessMiddleware::class, 'arguments' => array()),
```

Comment:
As you deal with the fact that `arguments` might not be present, shouldn't we remove it here?

Reply:
The config adds the default [] value, but not the default middleware in https://github.com/ogizanagi/symfony/blob/a36dad010b7634bbe3c35f74490ced841d0fc2a1/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L1472-L1474

Should I update the extension and enforce in the pass? Or the config so it removes it if empty? Not really worth it IMHO.

Reply:
NVM, understood. I thought this was the `FrameworkExtensionTest` class 😅